swarm/api/http/server_test.go

```diff
@@ -1153,3 +1155,123 @@ func TestMultiPartUpload(t *testing.T) {
 		t.Fatalf("expected POST multipart/form-data to return a 64 char manifest but the answer was %d chars long", len(body))
 	}
 }
+
+// TestBzzGetFileWithResolver tests fetching a file using a mocked ENS resolver
+func TestBzzGetFileWithResolver(t *testing.T) {
+	resolver := newTestResolveValidator("")
+	srv := testutil.NewTestSwarmServer(t, serverFunc, resolver)
+	defer srv.Close()
+	fileNames := []string{"dir1/tmp1.txt", "dir2/tmp2.lock", "dir3/tmp3.rtf"}
+	fileContents := []string{"tmp1textfilevalue", "tmp2lockfilelocked", "tmp3isjustaplaintextfile"}
+
+	buf := &bytes.Buffer{}
+	tw := tar.NewWriter(buf)
+	defer tw.Close()
```

Comment:
You probably want to close this tar writer before issuing the HTTP request so that `buf` contains the full tar archive including the footer (see [tar.Writer.Close](https://golang.org/pkg/archive/tar/#Writer.Close))